Update configuration.asciidoc (#1005)
elastic/apm-agent-ruby: docs/configuration.asciidoc

ifdef::env-github[]
NOTE: For the best reading experience,
please view this documentation at
https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html[elastic.co].
endif::[]

[[configuration]]
== Configuration

There are several ways to configure how Elastic APM behaves.

> We recommend the method of specifying options in a `config/elastic_apm.yml` file:

Reviewer: ```suggestion
We recommend using a `config/elastic_apm.yml` file:
```

[[configuration-precedence]]
=== Configuration precedence

> Options are applied in the following order (last one wins): 

Author: Should this say listed in descending order with the last taking precedence for consistency with other articles?

[[dynamic-configuration]]
=== Dynamic configuration

> Configuration options that are marked with the image:./images/dynamic-config.svg[] badge can be changed at runtime

Reviewer: ```suggestion
Configuration options marked with the image:./images/dynamic-config.svg[] badge can be changed at runtime
```

[float]
=== Sinatra and Rack

> When using Sinatra and Rack, configure when starting

Reviewer: ```suggestion
When using Sinatra and Rack, you can configure when starting
```

This string is used to ensure that only your agents can send data to your APM server.
Both the agents and the APM server have to be configured with the same secret token.

> An example to generate a secure secret token is:

Reviewer: ```suggestion
Here's an example that generates a secure secret token:
```

|============

This base64-encoded string is used to ensure that only your agents can send data to your APM server.

> The API key must be created using the APM server command line tool. Please see the APM server
> documentation for details on how to create the API key. Please note that this feature is experimental in the
> APM server in version 7.6.

Reviewer: ```suggestion
The API key must be created using the {apm-server-ref-v}/api-key.html[APM server command-line tool]. 
```

| `ELASTIC_APM_API_BUFFER_SIZE` | `api_buffer_size` | `256`
|============

The maximum amount of objects kept in queue before sending to APM Server.

If the limit is hit, increase the agent's
<<config-pool-size,worker pool size>> or the agent is having trouble

> connecting to APM Server. The <<config-log-path,logs>> should help to troubleshoot.

Author: Is this better than saying "should tell you what went wrong" ?

> If the limit is hit, increase the agent's
> <<config-pool-size,worker pool size>> or the agent is having trouble
> connecting to APM Server. The <<config-log-path,logs>> should help to troubleshoot.

Reviewer: ```suggestion
If you hit the limit, consider increasing the agent's <<config-pool-size,worker pool size>>.
If you don't, the agent may have trouble connecting to APM Server.
The <<config-log-path,logs>> should tell you what went wrong.
```

| `ELASTIC_APM_BREAKDOWN_METRICS` | `breakdown_metrics` | `true`
|============

Enable or disable the tracking and collection of breakdown metrics.

> By setting this to `False`, the tracking is disabled, which can reduce the overhead of the agent.

Reviewer: ```suggestion
Setting this to `False` disables the tracking of breakdown metrics, which can reduce the overhead of the agent.
```

| `ELASTIC_APM_CAPTURE_HEADERS` | `capture_headers` | `true`
|============

> This indicates whether or not to attach the request headers to transactions and errors.

Reviewer: ```suggestion
Configure whether or not to attach the request headers to transactions and errors.
```

| `ELASTIC_APM_CENTRAL_CONFIG` | `central_config` | `true`
|============

This enables {apm-app-ref}/agent-configuration.html[APM Agent Configuration via Kibana].
If set to `true`, the client will poll the APM Server regularly for new agent configuration.

> Usually APM Server determines how often to poll, but if not set the default interval is 5 minutes.

Reviewer: ```suggestion
Usually APM Server determines how often to poll, but if not, set the default interval is 5 minutes.
```

| `ELASTIC_APM_INSTRUMENTED_RAKE_TASKS` | `instrumented_rake_tasks` | `[]`    | `['my_task']`
|============

> Elastic APM can instrument your Rake tasks. Theis is an opt-in field, as they are used are for a multitude of things. 

Author: Does this make sense or is it changing the meaning too much?

Reviewer: ```suggestion
Elastic APM can instrument your Rake tasks. Because Rake tasks are used for very different things, this is an opt-in field.
```

| `ELASTIC_APM_SANITIZE_FIELD_NAMES` | `sanitize_field_names` | `"password,passwd,pwd,secret,*key,*token*,*session*,*credit*,*card*,authorization,set-cookie"` | `Auth*tion,abc*,*xyz`
|============

> Sometimes it is necessary to sanitize the data sent to Elastic APM to remove sensitive data.

Reviewer: ```suggestion
Sometimes it is necessary to sanitize the data sent to Elastic APM to remove sensitive values.
```

| `ELASTIC_APM_SERVICE_NAME` | `service_name` | App's name | `MyApp`
|============

> The name of your service. This is used to group the errors and transactions of your service and is the 

Reviewer: ```suggestion
The name of your service. This is used to group the errors and transactions of your service and is
```

the primary filter in the Elastic APM user interface.

If you're using Ruby on Rails this will default to your app's name.
If you're using Sinatra it will default to the name of your app's class.

NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]+$`.

> Your service name must only contain characters from the ASCII

Reviewer: ```suggestion
In other words, it must only contain characters from the ASCII
```

 * `mb` (megabytes)
 * `gb` (gigabytes)

NOTE: We use the power-of-two sizing convention, e.g. `1 kilobyte == 1024 bytes`.

[float]
[[special-configuration]]
=== Special configuration

> Elastic APM patches `Kernel#require` to auto-detect and instrument supported third party libraries. However, it may conflict with some libraries.

Reviewer: ```suggestion
Elastic APM patches `Kernel#require` to auto-detect and instrument supported third party libraries. It does so with the utmost care but in rare cases it can conflict with some libraries.
```
